src/java.desktop/share/classes/java/awt/color/ICC_Profile.java

```diff
@@ -995,13 +961,7 @@ public static ICC_Profile getInstance(String fileName) throws IOException {
      *         Profile data
      */
     public static ICC_Profile getInstance(InputStream s) throws IOException {
-    byte[] profileData;
-
-        if (s instanceof ProfileDeferralInfo) {
-            /* hack to detect profiles whose loading can be deferred */
```

Comment:
As of now, this code block is never executed, since we never pass ProfileDeferralInfo as an InputStream. probably we did it a long time ago.